test/unittest/fem/test_inversetransform.cpp

```diff
@@ -0,0 +1,170 @@
+   "1"                 "\n"
+   "1 3 0 1 2 3"     "\n\n"
+   "boundary"          "\n"
+   "0"               "\n\n";
+
+// Nodal grid function for a C-shaped quadratic quadrilateral
+std::string CShapedNodesStr =
+   "vertices"                            "\n"
+   "4"                                 "\n\n"
+   "nodes"                               "\n"
+   "FiniteElementSpace"                  "\n"
+   "FiniteElementCollection: Quadratic"  "\n"
+   "VDim: 2"                             "\n"
+   "Ordering: 1"                         "\n"
+   "0 0"                                 "\n"
+   "0 2"                                 "\n"
+   "0 6"                                 "\n"
+   "0 8"                                 "\n"
+   "0 1"                                 "\n"
+   "-6 4"                                "\n"
+   "0 7"                                 "\n"
+   "-8 4"                                "\n"
+   "-7 4"                                "\n";
+
+TEST_CASE("InverseElementTransformation",
+          "[InverseElementTransformation]")
+{
+   typedef InverseElementTransformation InvTransform;
+
+   // Create quadratic with single C-shaped quadrilateral
+   std::stringstream meshStr;
+   meshStr << meshPrefixStr << CShapedNodesStr;
+   Mesh mesh( meshStr );
+
+   REQUIRE( mesh.GetNE() == 1 );
+   REQUIRE( mesh.GetNodes() != NULL );
+
+   // Optionally, dump mesh to disk
+   bool dumpMesh = false;
+   if (dumpMesh)
+   {
+      std::string filename = "c_shaped_quadratic_mesh";
+      VisItDataCollection dataCol(filename, &mesh);
+      dataCol.Save();
+   }
+
+   const int res = 100;
+   const int dim = 2;
+   const double tol = 2e-14;
+
+   SECTION("{ C-shaped Q2 Quad }")
+   {
+      // Create a uniform grid of integration points over the element
+      const int geom = mesh.GetElementBaseGeometry(0);
+      RefinedGeometry* ref =
+         GlobGeometryRefiner.Refine(Geometry::Type(geom), res);
+      const IntegrationRule& intRule = ref->RefPts;
+
+      // Create a transformation
+      IsoparametricTransformation tr;
+      mesh.GetElementTransformation(0, &tr);
+      Vector v(dim);
+
+      const int npts = intRule.GetNPoints();
+      int pts_found = 0;
+      double max_err = 0.0;
+      for (int i=0; i<npts; ++i)
+      {
+         // Transform the integration point into space
+         const IntegrationPoint& ip = intRule.IntPoint(i);
+         tr.Transform(ip, v);
+
+         // Now reverse the transformation
+         IntegrationPoint ipRev;
+
+         int res = tr.TransformBack(v, ipRev);
+
+         // Check that the reverse transform was successful
+         if ( res == InvTransform::Inside )
+         {
+            pts_found++;
+
+            // Accumulate the maximal error
+            max_err = std::max(max_err, std::abs(ipRev.x - ip.x));
+            max_err = std::max(max_err, std::abs(ipRev.y - ip.y));
+         }
+      }
+      std::cout << "Points found: " << pts_found << '/' << npts << '\n'
```

Comment:
Do we want this output by default?

Reply:
Output like this does not affect the pass/fail of a test. So informational messages like these, that may help you debug issues on failure, are useful. I was actually looking exactly at this output recently, trying to figure out why a test failed.

Reply:
I'm inclined to see it that way too, but I wanted to hear what other folks think...

Reply:
I agree, chatty output helps find bugs when tests fail.  It might be best to only print the output if the test fails though, but that is a bit of a pain when developing the tests.